misc/co2_plume_spawner.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+from stl import mesh
+import numpy as np
+import math
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d.art3d import Poly3DCollection
+import random
+from skimage import measure
+from skimage.draw import ellipsoid
+
+ros1 = True
+try:
+  import rospy  # ros1
+except ModuleNotFoundError:
+  ros1 = False
+  import rclpy  # ros2
+  from gazebo_msgs.srv import SpawnEntity
+
+from geometry_msgs.msg import Point, Pose
+from gazebo_msgs.srv import SpawnModel
+
+"""
+To try it out try something like this
+docker exec -it `docker ps | grep dragonfly-sim | cut -d" " -f1` /bin/bash
+source /opt/ros/galactic/setup.bash
+cd workspace
+python3.8 ./misc/co2_plume_spawner.py
+"""
+
+THRESHOLD = 300  # ppm
+node = None
+spawn_model = None
+spawn_entity_client = None
+
+
+def plume(x, y, z, q=5000, k=2, u=1):
+  return (q / (2 * math.pi * k * x)) * np.exp(- (u * (pow(y, 2) + pow(z, 2)) / (4 * k * x)))
+
+
+def plume_grid():
+  x, y, z = np.mgrid[000.1:100:1, -20:20:1, 0:400:1]
+  return plume(x, y, z)
+
+
+plume_base = plume_grid()
```

Comment:
I wonder, could place the plume in gazebo with some input variables from the CLI?

Reply:
Yeah any defined model you can place(spawn) in gazebo from the command line

Reply:
You just gotta make sure it matches the virtual co2publisher or that we write something that looks at the current gazebo models